File: mindflow/utils/files/utf8.py
"""
Quick UTF-8 validation.
"""


def is_valid_utf8(document_text: str) -> bool:
    """
    Check if a file is valid utf8.
    """
    try:
        document_text.encode("utf-8")
        return True
    except UnicodeDecodeError:
        return False

File: mindflow/src/utils/files/utf8.rs
use simdutf8::basic::from_utf8;

// Take a file path and read the contents byte, by byte, if the file is valid utf8, return false, otherwise return true
pub fn is_valid_utf8(path: &str) -> bool {
    let text_bytes = match std::fs::read(path) {
        Ok(bytes) => bytes,
        Err(e) => {
            log::debug!("Could not read file: {}. error {:?}", path, e);
            return false
        }
    };
    match from_utf8(&text_bytes) {
        Ok(_text) => true,
        Err(_e) => false
    }
}